R/zchunk_L101.nonghg_en_USA_S_T_Y.R

```diff
@@ -0,0 +1,244 @@
+      select(-mode, -UCD_sector, -size.class, -UCD_technology, -UCD_fuel) %>%
+
+      # Bind all together
+      bind_rows(L1231.in_EJ_R_elec_F_tech_Yh, temp) ->
+      L101.in_EJ_R_en_Si_F_Yh
+
+    L101.in_EJ_R_en_Si_F_Yh %>%
+      gather(year, value, -GCAM_region_ID, -sector, -fuel, -technology) %>%
+      mutate(year = as.integer(substr(year, 2, 5))) ->
+      L101.in_EJ_USA_en_Sepa_F_Yh.mlt
+
+    # Subset for USA only and aggregate to EPA categories
+    GCAM_sector_tech %>%
+      select(EPA_agg_sector, EPA_agg_fuel, sector, fuel) %>%
+      distinct(sector, fuel, .keep_all = TRUE) ->
+      temp   # dataset we're about to merge below, replicating `match` behavior
+
+    L101.in_EJ_USA_en_Sepa_F_Yh.mlt %>%
+      filter(GCAM_region_ID == gcam.USA_CODE) %>%
+      left_join(temp, by = c("sector", "fuel")) %>%
+      group_by(EPA_agg_sector, EPA_agg_fuel, year) %>%
+      summarise(energy = sum(value)) %>%
+      filter(!is.na(EPA_agg_sector)) ->
+      L101.in_EJ_USA_en_Sepa_F_Yh.mlt
+
+    # Convert EPA SO2, CO, NOx, VOC, and NH3 emissions inventories to Tg and aggregate by sector and technology
+    # We do this for each gas, so define a function with all the steps
+    EPA_convert_and_aggregate <- function(x, EPA_tech) {
+      x %>%
+        gather(year, value, -Source_Category_Raw, -Source_Category) %>%
+        mutate(year = as.integer(year)) %>%
+        left_join(distinct(EPA_tech), by = c("Source_Category" = "EPA_Category")) %>%
+        filter(year %in% emissions.EPA_HISTORICAL_YEARS, !is.na(fuel)) %>%
+        group_by(sector, fuel, year) %>%
+        # summarise and convert to Tg
+        summarise(value = sum(value) * emissions.TST_TO_TG) %>%
+        # set missing values to zero
+        mutate(value = if_else(is.na(value), 0, value))
+    }
+
+    L101.so2_tg_USA_en_Sepa_F_Yh <- EPA_convert_and_aggregate(EPA_SO2, EPA_tech)
+    L101.co_tg_USA_en_Sepa_F_Yh <- EPA_convert_and_aggregate(EPA_CO, EPA_tech)
+    L101.nox_tg_USA_en_Sepa_F_Yh <- EPA_convert_and_aggregate(EPA_NOx, EPA_tech)
+    L101.voc_tg_USA_en_Sepa_F_Yh <- EPA_convert_and_aggregate(EPA_VOC, EPA_tech)
+    L101.nh3_tg_USA_en_Sepa_F_Yh <- EPA_convert_and_aggregate(EPA_NH3, EPA_tech)
+
+    # Compute SO2, CO, NOx, VOC, and NH3 emissions factors by dividing EPA inventory by IEA energy balances
+    # We do this for each gas, so define a function with all the steps
+    EPA_compute_emissions_factors <- function(x, L101.in_EJ_USA_en_Sepa_F_Yh.mlt) {
+      L101.in_EJ_USA_en_Sepa_F_Yh.mlt %>%
+        left_join(x, by = c("EPA_agg_sector" = "sector", "EPA_agg_fuel" = "fuel", "year" = "year")) %>%
+        rename(emissions = value) %>%
+        mutate(emissions = if_else(is.na(emissions), 0, emissions),
+               em_factor = emissions / energy,
+               em_factor = if_else(is.nan(em_factor) | is.infinite(em_factor), 0, em_factor)) ->
+        x_em_factor
+
+      # Compute sector emissions. If these are zero, then reset all fuel em_factors to 1
```

Comment:
Just out of curiosity, when sector emissions equal zero, why em_factors are set to 1 Tg/EJ?